From 2c6cfa74ea9def3e8b052dd407b332a1e841f865 Mon Sep 17 00:00:00 2001
From: Peter Chiu <chui-hao.chiu@mediatek.com>
Date: Fri, 23 Aug 2024 14:30:00 +0800
Subject: [PATCH 115/236] mtk: hostapd: fix wds remove flow

Fix the params set to set_wds_sta to prevent ifname_wds being null.
If ifname_wds is null, the wds interface cannot be remove.

Signed-off-by: Peter Chiu <chui-hao.chiu@mediatek.com>
---
 src/ap/ap_drv_ops.c          | 2 +-
 src/drivers/driver_nl80211.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ap/ap_drv_ops.c b/src/ap/ap_drv_ops.c
index d88b2b50b..50704b61a 100644
--- a/src/ap/ap_drv_ops.c
+++ b/src/ap/ap_drv_ops.c
@@ -396,7 +396,7 @@ int hostapd_set_wds_sta(struct hostapd_data *hapd, char *ifname_wds,
 			const u8 *addr, int aid, int val)
 {
 	const char *bridge = NULL;
-	char ifName[IFNAMSIZ + 1];
+	char ifName[IFNAMSIZ + 1] = {0};
 
 	int mRet = 0;
 
diff --git a/src/drivers/driver_nl80211.c b/src/drivers/driver_nl80211.c
index 28690fb0d..70cdb4412 100644
--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -8852,7 +8852,7 @@ static int i802_set_wds_sta(void *priv, const u8 *addr, int aid, int val,
 {
 	struct i802_bss *bss = priv;
 	struct wpa_driver_nl80211_data *drv = bss->drv;
-	char name[IFNAMSIZ + 1];
+	char name[IFNAMSIZ + 1] = {0};
 	union wpa_event_data event;
 	bool add_br = false;
 	int ret;
-- 
2.45.2

